chains/nomad-ethereum/src/macros.rs

```diff
@@ -106,61 +109,146 @@ macro_rules! boxed_indexer {
     };
 }
 
+/// Create base http retrying provider
+#[macro_export]
+macro_rules! http_provider {
+    ($url:expr) => {{
+        let provider: crate::retrying::RetryingProvider<ethers::providers::Http> = $url.parse()?;
+        Arc::new(ethers::providers::Provider::new(provider))
+    }};
+}
+
+/// Create base ws provider
+#[macro_export]
+macro_rules! ws_provider {
+    ($url:expr) => {{
+        let ws = ethers::providers::Ws::connect($url).await?;
+        Arc::new(ethers::providers::Provider::new(ws))
+    }};
+}
+
+/// Create ethers::SignerMiddleware from http connection
+#[macro_export]
+macro_rules! wrap_http {
+    ($provider:expr, $signer:ident) => {{
+        // First set the chain ID locally
+        let provider_chain_id = $provider.get_chainid().await?;
+        let signer = ethers::signers::Signer::with_chain_id($signer, provider_chain_id.as_u64());
+
+        // Manage the nonce locally
+        let address = ethers::prelude::Signer::address(&signer);
+        let provider =
+            ethers::middleware::nonce_manager::NonceManagerMiddleware::new($provider, address);
+
+        // Kludge. Increase the gas by multiplication of every estimated gas by
+        // 2, except the gas for chain id 1 (Ethereum Mainnet)
+        let provider = crate::gas::GasAdjusterMiddleware::with_default_policy(
+            provider,
+            provider_chain_id.as_u64(),
+        );
+
+        // Manage signing locally
+        Arc::new(ethers::middleware::SignerMiddleware::new(provider, signer))
+    }};
+}
+
+/// Create ethers::SignerMiddleware from websockets connection
+#[macro_export]
+macro_rules! wrap_ws {
```

Comment:
Are `wrap_rs` and `wrap_http` identical? If so let's merge them

Reply:
Yep thx good catch

Reply:
Merged to same method :)